scripts/hashlint.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+"""
+Simple script to validate the various hashes for downloads across the project.
+"""
+
+import hashlib
+import sys
+
+import requests
+import yaml
+
+
+URLS = {
+    "roles/jgrasp/vars/main.yml": {
+        "hash": "jgrasp.hash",
+        "urls": ["jgrasp.url"],
+    },
+    "roles/finch/vars/main.yml": {
+        "hash": "finch.hash",
+        "urls": ["finch.url"],
+    },
+    "roles/eclipse/vars/main.yml": {
+        "hash": "eclipse.hash",
+        "urls": ["eclipse.url", "eclipse.url_backup"],
+    },
+}
+
+
+def get_field(data, key):
+    """
+    Get a field separated by dots
+    """
+
+    if isinstance(key, str):
+        key = key.split('.')
+    else:
+        print(type(key))
+    while key:
+        data = data[key.pop(0)]
+    return data
+
+
+def check_software_hash(session, url, expected, file):
+    """
+    Checks the hash for the contents of a URL against an expected value
+    """
+
+    try:
+        data = session.get(url).content
```

Comment:
Maybe borrow Ansible's user-agent to appease Finch: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/get_url.py#L170